src/components/paint_format_button/paint_format_store.ts

```diff
@@ -64,27 +71,72 @@ export class PaintFormatStore extends SpreadsheetStore {
     return this.status !== "inactive";
   }
 
+  private get clipboardHandlers(): {
+    handlerName: string;
+    handler: ClipboardHandler<any>;
+  }[] {
+    if (!this._clipboardHandlers) {
+      this._clipboardHandlers = PAINT_FORMAT_HANDLER_KEYS.map((handlerName) => {
+        const Handler = clipboardHandlersRegistries.cellHandlers.get(handlerName);
+        return {
+          handlerName,
+          handler: new Handler(this.getters, this.model.dispatch),
+        };
+      });
+    }
+    return this._clipboardHandlers;
+  }
+
   private copyFormats(): ClipboardContent | undefined {
     const sheetId = this.getters.getActiveSheetId();
     const zones = this.getters.getSelectedZones();
+    const copiedData: Partial<ClipboardContent> = { zones, sheetId };
 
-    const copiedData = {};
-    for (const handler of this.clipboardHandlers) {
-      Object.assign(copiedData, handler.copy(getClipboardDataPositions(sheetId, zones)));
+    for (const { handlerName, handler } of this.clipboardHandlers) {
+      const handlerResult = handler.copy(getClipboardDataPositions(sheetId, zones));
+      if (handlerResult !== undefined) {
+        copiedData[handlerName] = handlerResult;
+      }
     }
 
     return copiedData as ClipboardContent;
   }
 
   private paintFormat(sheetId: UID, target: Zone[]) {
-    if (this.copiedData) {
-      for (const handler of this.clipboardHandlers) {
-        handler.paste({ zones: target, sheetId }, this.copiedData, {
-          isCutOperation: false,
-          pasteOption: "onlyFormat",
-        });
+    const selectedZones: Zone[] = [];
+    const copiedData = this.copiedData;
+    if (!copiedData) {
+      return;
+    }
+
+    for (const { handlerName, handler } of this.clipboardHandlers) {
```

Comment:
this whole thing is duplicated from the clipboard, would it be possible to factorize that into an helper you think ? Or maybe it's too annoying